src/pymor/analyticalproblems/functions.py

```diff
@@ -61,22 +61,73 @@ def __add__(self, other):
             if np.all(other == 0.):
                 return self
             other = ConstantFunction(other, dim_domain=self.dim_domain)
-        return LincombFunction([self, other], [1., 1.])
 
-    __radd__ = __add__
+        if self.name != 'LincombFunction':
+            if isinstance(other, LincombFunction) and other.name == 'LincombFunction':
+                functions = (self,) + other.functions
+                coefficients = (1.,) + (other.coefficients if sign == 1. else tuple(-c for c in other.coefficients))
+            else:
+                functions, coefficients = (self, other), (1., sign)
+        elif isinstance(other, LincombFunction) and other.name == 'LincombFunction':
+            assert isinstance(self, LincombFunction)
```

Comment:
This is a bit strange. If `other` has the name `'LincombFunction'` but is not a `LincombFunction` that's fine. If the same holds for `self` this is an error.

Reply:
So you'd like to give it an error also for the case that `other` has the name `LincombFunction` but is not one?

Reply:
Both versions would be fine with me, but addition should be commutative ...